addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -330,10 +330,11 @@ def _l10n_es_tbai_get_invoice_values(self, invoice, cancel):
         invoice_lines = []
         for line in invoice.invoice_line_ids.filtered(lambda line: line.display_type not in ('line_section', 'line_note')):
             if line.discount == 100.0:
-                gross_price_subtotal = line.price_unit * line.quantity
+                # no currency conversion since balance , amount_currency should be 0
+                balance_before_discount = - line.price_unit * line.quantity
             else:
-                gross_price_subtotal = line.price_subtotal / (1 - line.discount / 100.0)
-            discount = (gross_price_subtotal - line.price_subtotal) * refund_sign
+                balance_before_discount = line.balance / (1 - line.discount / 100)
```

Comment:
Your variable is not using the same currency in both logical branches

Reply:
The problem is: We need to determine the exchange rate of the line. It is i.e. used to convert between `balance` and `price_subtotal` (and not really anywhere else on stored values except for the sums). But here both values should be 0 due to the 100% discount, so any exchange rate could have been used to convert between the 0 and 0. So we can't know the exchange that was used.
It is the way we usually do it.
The new version should only fail to convert if the whole invoice has a 0 total
(since it basically uses the sum of all balance and sum of all price_subtotal).